pkg/microservice/warpdrive/core/service/taskplugin/job.go

```diff
@@ -1035,3 +1057,12 @@ func genRegistrySecretName(reg *task.RegistryNamespace) (string, error) {
 
 	return secretName, nil
 }
+
+func replaceEnvWithValue(str string, envs map[string]string) string {
```

Comment:
Here is a method rendering environment variable that we could consider pulling out for public use: https://github.com/koderover/zadig/blob/main/pkg/microservice/warpdrive/core/service/taskplugin/build.go#L473-L487

Reply:
The comment above says it is a temporary function, I'm not so sure about that.